### PR TITLE
Shifted away from (abandoned) Yahoo API to OpenWeatherMap

### PR DESCRIPTION
**Pull request by StopMotionCuber**
I made some efforts to make the Weather segment work again.
This adresses issue #1973
Downside of OpenWeatherMap is that it needs an API key, but they state [to welcome FOSS](https://openweathermap.org/faq) so I'm going to request an API key for this project there.

This is not really ready for merging yet, but already in a working state (if you provide an API key).
I'm not gonna lie, I submitted this PR early as hacktober is almost over and this is my last PR that I need for Hacktoberfest.

### CODE REVIEW

**Comment by ingorichter** [comment]
I'm interested in getting this for myself. What is the best way to get this PR merged?

**Comment by StopMotionCuber** [comment]
Bumping this issue was a good idea to get this MR rolling.
I think this can be merged now, although I did not take a too deep look into the icons associated with weather codes returned by OpenWeatherMap. Right now the icons selected are rather generic.

**Comment by ingorichter** [comment]
How do you get the icons? I see only the textual representation of the current weather. But it's fine for me. Did you consider mapping all different weather situations from https://openweathermap.org/weather-conditions? That would be a huge improvement in detail information provided. The PR looks good 👏

**Comment by StopMotionCuber** [comment]
The icons are applied by the skin based on the textual representation.
You can basically take a look at the diff for the `weather_conditions_codes` dict which names are available.
When there are multiple names, the skin will try to match the first name, in case there is no icon it will fall back to the generic name (at least that is my understanding).

File: powerline/segments/common/wthr.py [hunk 1]
```diff
@@ -105,6 +105,7 @@ class WeatherSegment(KwThreadedSegment):
 	interval = 600
 	default_location = None
 	location_urls = {}
+	weather_api_key = "fbc9549d91a5e4b26c15be0dbdac3460"
```
Comment (PH111P) [inline powerline/segments/common/wthr.py:108]: Perhaps I'm missing something here, but it seems wrong to me to have an API key hard-coded.

Comment (ingorichter) [reply powerline/segments/common/wthr.py:108]: that could well be in the config file :+1

**Comment by ingorichter** [comment]
@StopMotionCuber I was thinking Icon==Image

File: powerline/segments/common/wthr.py (hunk 1 above)
Comment (StopMotionCuber) [reply powerline/segments/common/wthr.py:108]: Yeah, I will  to make that work.

**Comment by StopMotionCuber** [comment]
> @StopMotionCuber I was thinking Icon==Image

I'm not sure what exactly you mean. As we are on the command line, images are generally not possible, at least not in a png way. This segment works with Unicode symbols/emojis.
If you mean something else please clarify that, as I'm not sure whether I'm getting your point.

**Comment by StopMotionCuber** [comment]
You can find one example for the unicode symbol config here:
https://github.com/powerline/powerline/blob/develop/powerline/config_files/themes/powerline_unicode7.json#L73

**Comment by ingorichter** [comment]
@StopMotionCuber :-) Yes, I was thinking symbols/emojis is the only way to show something on the command line, that's why I was wondering about the term icon in this context. Thanks for the explanations. I just wonder why I don't see the emojis in my setup 🤔 Instead of the clouds emoji I see **CLOUDS** at string instead

File: powerline/segments/common/wthr.py (hunk 1 above)
Comment (PH111P) [reply powerline/segments/common/wthr.py:108]: How about adding the api key as a parameter to the segment?

**Comment by StopMotionCuber** [comment]
I changed the structure a bit, now the API key can be changed and it's also documented how to change.
I took a look at the IMAP segment for this implementation

**Comment by PH111P** [comment]
Looks good! Thanks for the great work @StopMotionCuber!
I suppose this PR is now ready for merging (once the tests pass, which they will, I presume)

**Comment by StopMotionCuber** [comment]
Quick update on this, I just got a reply from the OpenWeatherMap support and the API key is now registered under opensource conditions (3000 API calls/min)

**Pull request merged**